src/test/scala/chiselTests/Mem.scala

```diff
@@ -512,40 +512,89 @@ class SRAMSpec extends ChiselFunSpec {
 
-  it(s"should support masking with Vec-valued data") {
-    class TestModule(val wr: Int, val rw: Int) extends Module {
-      val mem = SRAM.masked(32, Vec(3, UInt(8.W)), 0, wr, rw)
+    it(s"should support masking with Vec-valued data") {
+      class TestModule(val wr: Int, val rw: Int) extends Module {
+        val mem = SRAM.masked(32, Vec(3, UInt(8.W)), 0, wr, rw)
 
-      dontTouch(mem)
+        dontTouch(mem)
 
-      for (i <- 0 until wr) {
-        mem.writePorts(i) := DontCare
-      }
-      for (i <- 0 until rw) {
-        mem.readwritePorts(i) := DontCare
+        for (i <- 0 until wr) {
+          mem.writePorts(i) := DontCare
+        }
+        for (i <- 0 until rw) {
+          mem.readwritePorts(i) := DontCare
+        }
       }
-    }
-    val chirrtl = ChiselStage.emitCHIRRTL(new TestModule(1, 1), args = Array("--full-stacktrace"))
+      val chirrtl = ChiselStage.emitCHIRRTL(new TestModule(1, 1), args = Array("--full-stacktrace"))
 
-    chirrtl should include(
-      "writePorts : { flip address : UInt<6>, flip enable : UInt<1>, flip data : UInt<8>[3], flip mask : UInt<1>[3]}[1]"
-    )
-    chirrtl should include(
-      "readwritePorts : { flip address : UInt<6>, flip enable : UInt<1>, flip isWrite : UInt<1>, readData : UInt<8>[3], flip writeData : UInt<8>[3], flip mask : UInt<1>[3]}[1]"
-    )
-
-    for (i <- 0 until 3) {
-      chirrtl should include(s"when mem.writePorts[0].mask[$i]")
-      chirrtl should include(s"connect mem_MPORT[$i], mem.writePorts[0].data[$i]")
-
-      chirrtl should include(s"when mem.readwritePorts[0].mask[$i]")
       chirrtl should include(
-        s"connect mem_out_readwritePorts_0_readData_MPORT[$i], mem.readwritePorts[0].writeData[$i]"
+        "writePorts : { flip address : UInt<6>, flip enable : UInt<1>, flip data : UInt<8>[3], flip mask : UInt<1>[3]}[1]"
       )
+      chirrtl should include(
+        "readwritePorts : { flip address : UInt<6>, flip enable : UInt<1>, flip isWrite : UInt<1>, readData : UInt<8>[3], flip writeData : UInt<8>[3], flip mask : UInt<1>[3]}[1]"
+      )
+
+      for (i <- 0 until 3) {
+        chirrtl should include(s"when mem.writePorts[0].mask[$i]")
+        chirrtl should include(s"connect mem_MPORT[$i], mem.writePorts[0].data[$i]")
+
+        chirrtl should include(s"when mem.readwritePorts[0].mask[$i]")
+        chirrtl should include(
+          s"connect mem_out_readwritePorts_0_readData_MPORT[$i], mem.readwritePorts[0].writeData[$i]"
+        )
+      }
+    }
+
+    it(s"should support multiple clocks driving different ports") {
+      class TestModule extends Module {
+        val (counter, _) = Counter(true.B, 11)
+        val clock1: Clock = (counter === 1.U).asClock
+        val clock2: Clock = (counter === 2.U).asClock
+        val clock3: Clock = (counter === 3.U).asClock
+
+        val readClocks = Seq(clock1, clock2, clock3)
+        val writeClocks = Seq(clock2, clock3, clock1)
+        val readwriteClocks = Seq(clock3, clock1, clock2)
```

Comment:
This is fun but maybe too clever by half, it would be more straightforward to just have Vecs of clocks (which are Seqs)
```suggestion
        val readClocks = IO(Input(Vec(3, Clock())))
        val writeClocks = IO(Input(Vec(3, Clock())))
        val readwriteClocks = IO(Input(Vec(3, Clock())))
```
Then below you just look for the clocks, eg. `readClocks[$i]`